lib/zip/extra_field.rb

```diff
@@ -3,7 +3,7 @@ class ExtraField < Hash
     ID_MAP = {}
 
     def initialize(binstr = nil)
-      binstr and merge(binstr)
+      binstr && merge(binstr)
```

Comment:
If this code need `and` to implement control flow - I think it need to be redone without usage of `and`, because of https://github.com/bbatsov/ruby-style-guide#no-and-or-or

Also since this code needed exactlye 'and' I think there is need of test, that failed on change on `&&` since not a single test failed after all my commits